src/main/java/ProductController.java

```diff
@@ -0,0 +1,39 @@
+import org.springframework.web.bind.annotation.*;
+import java.util.*;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+    private final Map<Long, Product> products = new HashMap<>();
+    private long nextId = 1L;
+
+    @PostMapping
+    public Product create(@RequestBody ProductRequest request) {
+        Product product = new Product(nextId++, request.getName(), request.getPrice());
+        products.put(product.getId(), product);
+        return product;
+    }
+
+    @GetMapping("/{productId}")
+    public Product read(@PathVariable Long productId) {
+        return Optional.ofNullable(products.get(productId))
+                .orElseThrow(() -> new NoSuchElementException("상품을 찾을 수 없습니다."));
+    }
+
+    @PutMapping("/{productId}")
+    public Product update(@PathVariable Long productId, @RequestBody ProductRequest request) {
+        Product product = Optional.ofNullable(products.get(productId))
+                .orElseThrow(() -> new NoSuchElementException("상품을 찾을 수 없습니다."));
+        Product updated = new Product(productId, request.getName(), request.getPrice());
+        products.put(productId, updated);
+        return updated;
```

Comment:
새로운 객체를 만들어서 저장소의 객체만 바꿔끼우는 것이 아니라 [객체에게 메시지를 보내보면](https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/) 어떨까요? 😃

```suggestion
        Product product = Optional.ofNullable(products.get(productId))
                .orElseThrow(() -> new NoSuchElementException("상품을 찾을 수 없습니다."));
        product.update(request.getName(), request.getPrice());

        return product;
```

Reply:
객체에게 메시지를 보낸다는것이 무슨말인지 이해가 안 되었는데 
올려주신 링크 글 읽어보고 이해가 되었습니다.
getter, setter, 프로퍼티 사용을 지양하는 상황과도 연관되어 있음을 이해했습니다.
수정해보겠습니다 !

Reply:
잘 이해가 되셨다니 다행이네요. 
아마도 객체지향 프로그래밍이 익숙하지 않으실거라고 생각해요. 😃
이번 기회에 조금씩 객체지향 프로그래밍에 다가가보면 좋을 것 같네요. 😉